drop-boxes/register-mtb-projects-dropbox/register-mtb-data-dropbox.py

```diff
@@ -0,0 +1,758 @@
+   if len(fastqs_tumor) != 2 or len(fastqs_normal) != 2:
+        raise mtbutils.MTBdropboxerror("Tumor/normal fastq dataset was not complete. Please check.")
+   else:
+        proc_fastq(fastqs_tumor, transaction)
+        proc_fastq(fastqs_normal, transaction)
+
+
+def execute_rnaseq_registration(rna_seq_files, transaction):
+    register_rnaseq(rna_seq_files, transaction)
+
+
+def get_last_exp_id(experiments):
+    """Fetches the highest experiment number from a list of experiments and returns its number."""
+    exp_ids = [int(re.search(r'[E](\d*)$', oid.getExperimentIdentifier()).group(1)) for oid in experiments if re.search(r'[E](\d*)$', oid.getExperimentIdentifier())]
+    exp_ids.sort()
+    return exp_ids[-1]
+
+
+def getNextFreeBarcode(projectcode, numberOfBarcodes):
+    letters = string.ascii_uppercase
+    numberOfBarcodes += 1
+
+    currentLetter = letters[numberOfBarcodes / 999]
+    currentNumber = numberOfBarcodes % 999
+    code = projectcode + str(currentNumber).zfill(3) + currentLetter
+    return code + checksum.checksum(code)
+
+
+def register_rnaseq(rna_seq_files, transaction):
+    """Registers RNAseq experiment raw data in openBIS.
+
+    The list must contain two elements, following the naming convention ``r'.*tumor_rna[1,2]{1}.fastq.gz'``.
+    Both files must additionally contain the same QBiC sample code in order to get registered.
+
+    Args:
+        rna_seq_files (list): A list with fastq files from an RNAseq experiment
+        transaction (:class:DataSetRegistrationTransaction): An openBIS data set registration object
+
+    Raises:
+        MTBdropboxerror: If some of the conditions have not been fullfilled, with a text string explaining
+                        the reason for the failure.
+    """
+    print(mtbutils.log_stardate('Registering incoming MTB RNAseq data {}'.format(rna_seq_files)))
+    assert len(rna_seq_files) == 2
+    file1 = os.path.basename(rna_seq_files[0])
+    file2 = os.path.basename(rna_seq_files[1])
+    assert len(set(QCODE_REG.findall(file1))) == 1
+    assert len(set(QCODE_REG.findall(file2))) == 1
+    assert QCODE_REG.findall(file1)[0] == QCODE_REG.findall(file2)[0]
+
+    # This is the tumor dna sample barcode (type: TEST_SAMPLE)
+    dna_barcode = QCODE_REG.findall(file1)[0]
+    # Find the corresponding space and project
+    space, project = space_and_project(dna_barcode)
+
+    search_service = transaction.getSearchService()
+    sc = SearchCriteria()
+    pc = SearchCriteria()
+    pc.addMatchClause(SearchCriteria.MatchClause.createAttributeMatch(SearchCriteria.MatchClauseAttribute.PROJECT, project));
+    sc.addSubCriteria(SearchSubCriteria.createExperimentCriteria(pc))
+    result = search_service.searchForSamples(sc)
+    print("Found {} samples for project {} in space {}.".format(len(result), project, space))
+    new_rna_sample_barcode = getNextFreeBarcode(project, numberOfBarcodes=len(result))
```

Comment:
this includes entity barcodes, which will lead to a miscount and future inconsistencies. it is better to check in a while loop if a certain code is already used (by fetching that sample) and incrementing the code as it is done in other ETL scripts.

Reply:
I am happy for a url with a cross-reference :)